### PR TITLE
Include submitted query and provider in “result-selection-ready” event

### PR DESCRIPTION
**Pull request by tiberio-baptista**
This PR adds the submitted query and provider to the `result-selection-ready` event as a new `request` field.

This further enriched the data that `result-batch` web plugins have access to.

### CODE REVIEW

File: dicoogle/src/main/resources/webapp/js/stores/requestStore.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  get: function() {
+    return {
+      contents: this._contents
+    };
+  }
+});
+
+export { RequestStore };
+
+window.store = RequestStore;
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/stores/requestStore.js:28]: Is this line needed?

File: dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/search/SearchServlet.java [hunk 2]
```diff
@@ -228,6 +228,7 @@ public void onReceive(Task<Iterable<SearchResult>> e) {}
                     JSONObject obj = dimModel.getJSONObject(depth, offset, psize);
                     elapsedTime = System.currentTimeMillis() - elapsedTime;
                     obj.put("elapsedTime", elapsedTime);
+                    obj.put("query", query);
```
[the remark this answers is not in the document]
Comment (tiberio-baptista) [reply dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/search/SearchServlet.java:231]: Forgot to include in commit. Nicely spotted.

File: dicoogle/src/main/resources/webapp/js/stores/requestStore.js (hunk 1 above)
Comment (tiberio-baptista) [reply dicoogle/src/main/resources/webapp/js/stores/requestStore.js:28]: Not needed, and neither was the trigger above. Removed both.

**Review by bastiao — APPROVED** [review]
Verified with an use case plugin and it works fine.

**Pull request merged**